habitat-lab/habitat/isaacsim/scene_instance_json_to_usd.py

```diff
@@ -0,0 +1,562 @@
+"""
+
+import argparse
+import asyncio
+import json
+import os
+import re
+from typing import Dict, List, Optional
+
+
+def sanitize_usd_name(name: str) -> Optional[str]:
+    """Sanitizes a string for use as a USD node name.
+
+    :param name: The input string to sanitize.
+
+    :return: The sanitized string, suitable for use as a USD node name.
+    """
+
+    # assert len(name) > 0
+    if len(name) == 0:
+        print("Input string for USD node is empty")
+        return None
+
+    # Replace spaces, hyphens, colons, and other special characters with underscores
+    sanitized_name = re.sub(r"[^\w]", "_", name)
+
+    # Remove consecutive underscores
+    sanitized_name = re.sub(r"_+", "_", sanitized_name)
+
+    # Ensure the name doesn't start with a number
+    if sanitized_name[0].isdigit():
+        sanitized_name = "_" + sanitized_name
+
+    return sanitized_name
+
+
+def object_usd_info(obj_instance_json, object_counts):
+    """From the values defined in the object json from the scence instance, generate various
+    strings and object counts.
+    :param obj_instance_json: This is object level information from the scence_instance json file, Dictionary object
+    :param object counts: This dictionary contains information about multiples uses of the same mesh
+
+    :return: object_config_filename is the name of the object config file.
+    out_usd_path is the filepath where the converted object is located. object_counts might be updated if there is another
+    instance of the same mesh. unique_object_name is the unique name of a mesh.
+    """
+    object_config_filename = (
+        f"{obj_instance_json['template_name']}.object_config.json"
+    )
+    # TODO: maybe don't prepend OBJECT_. Just use sanitize_usd_name.
+    # NOTE: You cannot start the unique object name with number in teh Xform class environment
+    base_object_name = "OBJECT_" + object_config_filename.removesuffix(
+        ".object_config.json"
+    )
+    base_object_name = sanitize_usd_name(base_object_name)
+
+    if base_object_name in object_counts:
+        object_counts[base_object_name] += 1
+        unique_object_name = (
+            f"{base_object_name}_dup{object_counts[base_object_name]}"
+        )
+    else:
+        unique_object_name = base_object_name
+        object_counts[base_object_name] = 1
+
+    out_usd_path = f"./data/usd/objects/{base_object_name}.usda"
+
+    return (
+        object_config_filename,
+        out_usd_path,
+        object_counts,
+        unique_object_name,
+    )
+
+
+############################################################################################
+# Convert scence instance object to a usd file
+############################################################################################
+
+
+def convert_object_to_usd(
+    objects_folder: str,
+    object_config_filename: str,
+    out_usd_path: str,
+    project_root_folder: str,
+) -> None:
+    """This converts an hssd object to a usda file.
```

Comment:
We shouldn't be specific to HSSD or name HSSD. (Although I'm okay with parameter default values being HSSD-centric.)

Reply:
Just bumping this comment. Let's not name HSSD in comments.